tests/sflow/test_sflow.py

```diff
@@ -56,13 +58,22 @@ def setup(duthosts, rand_one_dut_hostname, ptfhost, tbinfo, config_sflow_feature
 
     config_dut_ports(duthost, var['test_ports'][0:2], vlan=1000)
 
-    for port_channel, interfaces in list(mg_facts['minigraph_portchannels'].items()):
-        for port in interfaces['members']:
-            var['sflow_ports'][port] = {}
-            var['sflow_ports'][port]['ifindex'] = get_ifindex(duthost, port)
-            var['sflow_ports'][port]['port_index'] = get_port_index(duthost, port)
-            var['sflow_ports'][port]['ptf_indices'] = mg_facts['minigraph_ptf_indices'][port]
-            var['sflow_ports'][port]['sample_rate'] = SFLOW_RATE_DEFAULT
+    upstream_ports = []
+    if len(mg_facts["minigraph_portchannels"]) == 0:
+        topo = tbinfo["topo"]["type"]
+        upstream_neigh_type = get_upstream_neigh_type(topo)
+        upstream_ports = get_neighbor_port_list(duthost, upstream_neigh_type)
+    else:
+        for port_channel, interfaces in list(mg_facts['minigraph_portchannels'].items()):
+            upstream_ports.extend(interfaces['members'])
+
```

Comment:
Thanks @nhe-NV, I believe there could be other tests that might require this change.

Reply:
Here is the logic for collect the ports for this test, and if other test case has same logic, then it should also require such, for now I did not define a seperate function for it. When we find some other test also need it, then we can define the function in the common place.